### ISSUE TITLE
Change event not working

### ISSUE THREAD

**Issue by fpinela8** [minhur/bootstrap-toggle#178]
I'm trying to use Bootstrap-toggle within an angular application but the change event is not working.

`<input type="checkbox" data-toggle="toggle" data-on="ON" data-off="OFF" data-onstyle="success" (change)="doSomething()">`

If I do not include bootstrap-toggle libs (min.js + min.css) the change event is triggered successfully and the method invoked. As soon as I load the bootstrap-toggle to transform my input checkbox into a switch, the event is not triggered anymore event though if the switch is displayed/behaves correctly.

Any ideas?

bootstrap 4
jquery 3.2.1
angular 4

**Comment by markskayff** [comment]
Yes, happening to me too. The change event is being blocked here too. Cannot make it work as docs read.

**Comment by TobiasBauer** [comment]
I think the solution to this problem is, to put the 
`<script src="/js/bootstrap-toggle.min.js"></script>`
after the reference of the general jquery and your script

**Comment by chud37** [comment]
Did anyone fix this? I'm getting the same thing.  The toggle displays and works correctly, however when inspecting the document, the checkbox is not checked when it should be.  

I'm loading the bootstrap-toggle-min.js at the end of the document.

**Comment by natecox** [comment]
I'm having this issue as well, I think the problem may be that the event listener for the `div.toggle` container's click isn't bubbling, and it's blocking access to the native DOM listeners. 

https://github.com/minhur/bootstrap-toggle/blob/master/js/bootstrap-toggle.js#L177

**Comment by vincent-olivert-riera** [comment]
@minhur any chance to get this fixed?

**Comment by philsmy** [comment]
I think this is also why I cannot get it to work with Stimulus.js. No event shows up.

**Comment by taimazB** [comment]
Same problem here.  I used onchange="function()" in the html code instead, and it works.
